### PR TITLE
Features/upgrade7/steps sidebar

### PR DESCRIPTION
**Pull request by bisht-richa**
- Steps Sidebar implementation
- Steps factory to concentrate all upgrade steps state and functions
- Unit test
- Update modules and dependencies

### CODE REVIEW

File: assets/app/data/crowbar/steps.factory.js [hunk 1]
```diff
@@ -4,72 +4,94 @@
...
+        function refeshStepsList() {
+            var currentState = $state.current.name,
+                isCompletedStep = true;
+
+            for (var i = 0; i < factory.steps.length; i++) {
+
+                /** 
+                Update factory steps.active and steps.enabled to ture, 
+                if the steps.state is equal to the current state 
+                */
```
Comment (santiph) [inline assets/app/data/crowbar/steps.factory.js]:
```
                /** 
                 * Update active and enabled properties to true for the current state 
                 */
```

File: assets/app/data/crowbar/steps.factory.js [hunk 2]
```diff
@@ -4,72 +4,94 @@
...
+                if (factory.steps[i].state === currentState) {
+                    // Update factory.activeStep with the active step object
+                    factory.activeStep = factory.steps[i];
+                    factory.steps[i].active = true;
+                    factory.steps[i].enabled = isCompletedStep;
+                    isCompletedStep = false;
+
+                } else {
```
Comment (santiph) [inline assets/app/data/crowbar/steps.factory.js]: // The rest of the steps should not be active

**Comment by Itxaka** [comment]
LGTM +1

**Comment by santiph** [comment]
Looks good to me. Please @Itxaka @skazi0 could you review my commit for the directive's unit tests?
https://github.com/crowbar/crowbar-ui/pull/71/commits/3dc615c6b8867b33a697f3f4a2b813a3193e0e40

We're still missing unit test for the steps factory.

**Comment by Itxaka** [comment]
Some of the commits have no info about whats in the commit:

`Merge branch 'features/upgrade7/steps-sidebar' of https://github.com/crowbar/crowbar-ui into features/upgrade7/steps-sidebar`

Can they be fixed to actually say what was changed/added/removed please?

Would also be good to merge https://github.com/crowbar/crowbar-ui/pull/74 with this so we can get full testing of the upgrade.controller.js

File: assets/app/widgets/crowbar-upgrade-steps/crowbar-upgrade-steps.directive.spec.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
+/*global bard assert $rootScope $compile $httpBackend should */
+describe('Crowbar Upgrade Steps Directive', function () {
+
+    var directiveElement = 123,
```
Comment (skazi0) [inline assets/app/widgets/crowbar-upgrade-steps/crowbar-upgrade-steps.directive.spec.js]: why initialize this variable?

File: assets/app/widgets/crowbar-upgrade-steps/crowbar-upgrade-steps.directive.spec.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+            unorderedList = directiveElement.find('ul');
+            listItems = unorderedList.find('li'),
+            lastListItem = listItems.splice(listItems.length - 1, 1);
+        });
+
+        it('should contain a list of steps', function() {
+
+            expect(unorderedList.length).toEqual(1);
+            expect(listItems.length).toEqual(2);
```
Comment (skazi0) [inline assets/app/widgets/crowbar-upgrade-steps/crowbar-upgrade-steps.directive.spec.js]: maybe compare to size of steps list in scope?

**Review by skazi0 — CHANGES_REQUESTED** [review]
nitpicks

File: assets/app/widgets/crowbar-upgrade-steps/crowbar-upgrade-steps.directive.spec.js (hunk 3 above)
Comment (santiph) [reply assets/app/widgets/crowbar-upgrade-steps/crowbar-upgrade-steps.directive.spec.js]: Oops. Needs to be removed.

File: assets/app/widgets/crowbar-upgrade-steps/crowbar-upgrade-steps.directive.spec.js (hunk 4 above)
Comment (santiph) [reply assets/app/widgets/crowbar-upgrade-steps/crowbar-upgrade-steps.directive.spec.js]: Good catch. will update

**Review by skazi0 — APPROVED** [review]

**Review by Itxaka — APPROVED** [review]

File: assets/app/features/upgrade/upgrade.module.js [hunk 5]
```diff
@@ -2,5 +2,5 @@
     'use strict';
 
     angular
-        .module('crowbarApp.upgrade', []);
+        .module('crowbarApp.upgrade', ['ui.router', 'pascalprecht.translate', 'suseData']);
```
Comment (Itxaka) [inline assets/app/features/upgrade/upgrade.module.js:5]: Question: where is `pascalprecht.translate` coming from and being used for? Is this different than `$translate`?

**Review by Itxaka — APPROVED** [review]
Still looking good to me!

**Comment by bisht-richa** [comment]
Looks good 👍

File: assets/app/features/upgrade/upgrade.module.js (hunk 5 above)
Comment (santiph) [reply assets/app/features/upgrade/upgrade.module.js:5]: `pascalprecht.translate` is the module where the `$translate` resource is contained.
You can check https://angular-translate.github.io/ or https://angular-translate.github.io/docs/#/guide for more details, or ask me if you have more questions. :)

**Comment by santiph** [comment]
[(2) S49P21 #5 Layout - Steps Sidebar](https://trello.com/c/HHfMzKPR/103-2-s49p21-5-layout-steps-sidebar)

**Pull request merged**